weldx/geometry.py

```diff
@@ -99,255 +98,404 @@ def _to_list(var) -> list:
+        p = self._series.evaluate(**{self._series.position_dim_name: positions})
+        return p.data_array.transpose(..., "c").data
 
     @property
-    @UREG.wraps(_DEFAULT_LEN_UNIT, (None,), strict=True)
     def length(self) -> pint.Quantity:
-        """Get the segment length.
+        """Get the length of the segment."""
+        return self._length
 
-        Returns
-        -------
-        pint.Quantity
-            Segment length
 
-        """
-        return self._length
+# DynamicShapeSegment ---------------------------------------------------------
+
+
+class DynamicShapeSegment(DynamicBaseSegment):
+    """Shape segment class to define arbitrary 2d shapes."""
+
+    @property
+    @UREG.wraps(_DEFAULT_LEN_UNIT, (None,), strict=True)
+    def point_start(self) -> pint.Quantity:
+        """Get the starting point of the segment."""
+        return self.get_points(0)[0, :2]
 
     @property
     @UREG.wraps(_DEFAULT_LEN_UNIT, (None,), strict=True)
     def point_end(self) -> pint.Quantity:
-        """Get the end point of the segment.
+        """Get the end point of the segment."""
+        return self.get_points(self._max_coord)[0, :2]
+
+    @UREG.wraps(None, (None, _DEFAULT_LEN_UNIT), strict=True)
+    def rasterize(self, raster_width: pint.Quantity) -> pint.Quantity:
+        """Get an array of discrete raster points of the segment.
+
+        Parameters
+        ----------
+        raster_width:
+            The desired distance between two raster points. The actual distance will be
+            the closest possible value to the desired one that guarantees an equal
+            distance between all raster points.
 
         Returns
         -------
-        pint.Quantity
-            End point
+        pint.Quantity:
+            Array of raster points
 
         """
-        return self._points[:, 1]
+        if raster_width <= 0:
+            raise ValueError("'raster_width' must be a number greater than 0.")
+        num_pts = int(np.round(self._length.to(_DEFAULT_LEN_UNIT).m / raster_width)) + 1
+        num_pts = max(num_pts, 2)
 
-    @property
-    @UREG.wraps(_DEFAULT_LEN_UNIT, (None,), strict=True)
-    def point_start(self) -> pint.Quantity:
-        """Get the starting point of the segment.
+        vals = np.array([i / (num_pts - 1) for i in range(num_pts)])
+        vals[-1] = 1.0
+
+        return self.get_points(vals * self._max_coord)[:, :2].transpose()
+
+    @UREG.check(None, _DEFAULT_LEN_UNIT)
+    def apply_translation(self, vector: pint.Quantity) -> DynamicShapeSegment:
+        """Apply a translation in place.
+
+        Parameters
+        ----------
+        vector:
+            The translation vector.
 
         Returns
         -------
-        pint.Quantity
-            Starting point
+        DynamicShapeSegment:
+            A self-reference to the modified segment
 
         """
-        return self._points[:, 0]
+        vector = Q_([*vector.m, 0], _DEFAULT_LEN_UNIT)
```

Comment:
```suggestion
        vector = np.append(vector, 0)
```
Since `@UREG.check` only checks dimensionality we should avoid `.magnitude` (without `.to()`)